### ISSUE TITLE
Support autocomplete by partial charaters or wordings like initial of wordings

### ISSUE THREAD

**Issue by tom10271** [hashicorp/vscode-terraform#1177]
## Versions
v2.23.0

### Extension
<img width="511" alt="Screenshot 2022-07-13 at 10 18 46 AM" src="https://user-images.githubusercontent.com/5089629/178636886-c8de1df1-c8f9-46cb-b1ff-87382422d393.png">

```

```

### VS Code

```
Version: 1.69.0
Commit: 92d25e35d9bf1a6b16f7d0758f25d48ace11e5b9
Date: 2022-07-07T05:28:46.408Z (5 days ago)
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin arm64 21.5.0
```

### Operating System

<img width="213" alt="Screenshot 2022-07-13 at 10 19 31 AM" src="https://user-images.githubusercontent.com/5089629/178636946-ee938ef4-34fd-493e-af8f-1cf73b8abf39.png">

### Terraform Version

```
Terraform v1.2.4
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.27.0
```

**Comment by radeksimko** [comment]
Hi @tom10271 
It is true that we (only) use exact match when filtering the completion items [today](https://github.com/hashicorp/hcl-lang/search?q=HasPrefix).

Just to confirm, am I right you expect the completion items to be filtered using fuzzy search, for example what gopls provides?
![Screenshot 2022-07-13 at 11 26 22](https://user-images.githubusercontent.com/287584/178714189-cad83ef1-96e1-4ae6-adad-a9a10c1c1965.png)

**Comment by tom10271** [comment]
Yes correct. I mainly use JetBrains’ IDE and such feature speed up coding a lot.

**Comment by radeksimko** [comment]
Right - maybe it doesn't match the strict definition of fuzzy search, I meant more "loose matching", as opposed to exact matching.

Either way that is something that looks useful and something we could implement. Thanks for the suggestion.

**Comment by radeksimko** [comment]
> There is another issue:
> 
> When I remove a few characters, it should refresh the list. Currently I need to remove everything before it started to refresh again and I cannot type anything wrong, otherwise it will not work until I remove everything again.
> 
> Should I create another issue for it?

Yes please - if you can create a separate one, and ideally attach a screen recording, that would be amazing! It sounds like a bug and something we could address more easily, compared to implementing loose matching of candidates.

**Comment by tom10271** [comment]
Yes correct. I mainly use JetBrains’ IDE and such feature speed up coding a
lot.

「Radek Simko ***@***.***>」在 2022年7月13日 週三，下午6:37 寫道：

**Comment by FWest98** [comment]
Please add support for this. It is the main reason I dislike working on Terraform stuff in VS Code, because I keep having to type the entirety of `openstack_networking_` when I just want to add a simple `secgroup`, for example. Similarly for references, having to type out the entire resource type is very annoying.

**Comment by ddmedvedev** [comment]
I want to join the discussion. Compared to JetBrains IDEs, Terraform autocomplete in VS Code feels weaker

<img width="499" height="126" alt="Image" src="https://github.com/user-attachments/assets/4d201dfd-0cc5-44e9-9629-4b40017f57a8" />

<img width="510" height="304" alt="Image" src="https://github.com/user-attachments/assets/b2f12359-2e3f-45c3-bad0-0850c6903e57" />
